static_analyzer/engine/lsp_recycler.py

```
@@ -0,0 +1,151 @@
+        self._lsp.restart()
+        # Blocks until the workspace is loaded again: a workspace-backed server
+        # cannot answer a documentSymbol until it has read the project files.
+        self._lsp.document_symbol(self._probe_file, timeout=self._probe_timeout)
+        # A loaded workspace is not a warm reference index — that is built on the
+        # first references query. Pay for it here, on a request we can give a long
+        # timeout, rather than losing a whole batch of real positions to the
+        # per-request deadline. Measured: recycling without this tripled the
+        # "Timeout waiting for references request" warnings.
+        try:
+            self._lsp.references(self._probe_file, 0, 0, timeout=self._probe_timeout)
+        except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)